### PR TITLE
Add hostlist expand as alternative to scontrol

### LINKED ISSUE #2051: Alternative to the command scontrol
## 🚀 Feature

Using `idist` on a slurm cluster relies on the command `scontrol` for selecting an hostname as `MASTER_ADDR`.

https://github.com/pytorch/ignite/blob/86cec1165441a945c93c938c23544ac52a17a170/ignite/distributed/comp_models/native.py#L258

As discussed in #2048, we should provide another way whether the command missing, for instance, using container launcher as `pyxis`.

### PR DESCRIPTION
**Pull request by sdesrozis**
Fixes #2051 

Description:

Add an alternative to scontrol for expanding hostname list used by SLURM.

```python
hostnames = expand_hostlist("quartz[4-8]")
print(hostnames)

> ['quartz4', 'quartz5', 'quartz6', 'quartz7', 'quartz8']
```

Based on https://github.com/LLNL/py-hostlist

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/utils.py [hunk 1]
```diff
@@ -271,3 +272,70 @@ def wrapper(*args: Any, **kwargs: Dict[str, Any]) -> Callable:
         return cast(F, wrapper)
 
     return decorator
+
+
+def expand_hostlist(nodelist: str) -> List[str]:
```
Comment (vfdev-5) [inline ignite/utils.py:277]: I'm not sure if we need this in `ignite.utils` and as a public method. Let's keep it near in `ignite/distributed/comp_models/native.py` and as a private method...

**Review by vfdev-5 — COMMENTED** [review]
@sdesrozis thanks for the PR

File: ignite/utils.py (hunk 1 above)
Comment (sdesrozis) [reply ignite/utils.py:277]: Ok 👍🏻

**Review by vfdev-5 — APPROVED** [review]
Thanks @sdesrozis !

**Pull request merged**